### PR TITLE
core(byte-efficiency): mark n/a if no network records in timespan

### PR DESCRIPTION
**Pull request by adamraine**
This resolves most of the errors we were seeing for a timespan run after page load.

The solution is to mark all BE audits N/A if there were no network records in timespan mode. Conceptually, this makes some sense since ms savings are calculated from throughput in timespan mode. 

#### Possible improvement:

Not all BE audits require network records. It was `LoadSimulator`, used to calculate throughput savings for all BE audits, which requires network records. Instead of always returning N/A, we could get throughput from the settings when using DT or simulated throttling, but not provided throttling. Provided throttling would need to be N/A without network records in timespan mode.

### CODE REVIEW

File: lighthouse-core/test/fraggle-rock/api-test-pptr.js [hunk 1]
```diff
@@ -174,11 +174,11 @@ describe('Fraggle Rock API', () => {
       const {auditResults, erroredAudits, notApplicableAudits} = getAuditsBreakdown(result.lhr);
       expect(auditResults.length).toMatchInlineSnapshot(`50`);
 
-      expect(notApplicableAudits.length).toMatchInlineSnapshot(`9`);
+      expect(notApplicableAudits.length).toMatchInlineSnapshot(`22`);
       expect(notApplicableAudits.map(audit => audit.id)).toContain('server-response-time');
 
       // TODO(FR-COMPAT): Reduce this number by handling the error, making N/A, or removing timespan support.
-      expect(erroredAudits.length).toMatchInlineSnapshot(`14`);
+      expect(erroredAudits.length).toMatchInlineSnapshot(`1`);
```
Comment (patrickhulce) [inline lighthouse-core/test/fraggle-rock/api-test-pptr.js:181]: nice! what's the final one?

File: lighthouse-core/audits/byte-efficiency/byte-efficiency-audit.js [hunk 2]
```diff
@@ -114,11 +114,24 @@ class UnusedBytes extends Audit {
       devtoolsLog,
       settings,
     };
-
     const networkRecords = await NetworkRecords.request(devtoolsLog, context);
+
+    // Requesting load simulator requires non-empty network records.
+    // Timespan is not guaranteed to generate network records, so mark N/A if empty.
```
Comment (patrickhulce) [inline lighthouse-core/audits/byte-efficiency/byte-efficiency-audit.js:120]:
```suggestion
    // Timespans are not guaranteed to have any network activity.
    // There are no bytes to be saved if no bytes were downloaded, so mark N/A if empty.
```

WDYT? just to make super clear it's the logical conclusion not a technical limitation :)

**Review by patrickhulce — APPROVED** [review]
nice work!

File: lighthouse-core/test/fraggle-rock/api-test-pptr.js (hunk 1 above)
Comment (adamraine) [reply lighthouse-core/test/fraggle-rock/api-test-pptr.js:181]: `long-tasks`. It's not a byte efficiency audit but requests the page dependency graph, which throws an error when there are no network records. It requires some additional investigation before we decide to just mark as N/A.

**Pull request merged**